src/transformers/models/rag/modeling_rag.py

```diff
@@ -891,7 +912,12 @@ def generate(
         model_kwargs["num_return_sequences"] = num_beams
         model_kwargs["attention_mask"] = None
 
-        for index in range(len(input_ids)):
+        if input_ids is not None:
+            len_input_ids = len(input_ids)
```

Comment:
better to use `input_ids.shape[0]` here. Also IMO, `len_input_ids` is not a great name because most people would think it refers to the sequence length, but here we are looking for the `batch_size` if I understand correctly -> so I think 

```python 
batch_size = input_ids.shape[0] if input_ids is not None else context_input_ids.shape[0] // n_docs
```